twitter/Views/TweetCell.m

```diff
@@ -7,18 +7,130 @@
 //
 
 #import "TweetCell.h"
+#import "APIManager.h"
+#import "DateTools.h"
+
+@interface TweetCell()
+
+@property (weak, nonatomic) IBOutlet UIImageView *profilePicture;
+@property (weak, nonatomic) IBOutlet UILabel *userDisplayName;
+@property (weak, nonatomic) IBOutlet UILabel *userAccountName;
+@property (weak, nonatomic) IBOutlet UILabel *displayDate;
+@property (weak, nonatomic) IBOutlet UITextView *tweetText;
+
+
+@property (weak, nonatomic) IBOutlet UIButton *retweetButton;
+@property (weak, nonatomic) IBOutlet UILabel *retweetCountLabel;
+@property (weak, nonatomic) IBOutlet UIButton *favoriteButton;
+@property (weak, nonatomic) IBOutlet UILabel *likeCountLabel;
+@property (weak, nonatomic) IBOutlet UIButton *replyButton;
+@property (weak, nonatomic) IBOutlet UILabel *replyCountLabel;
+
+@end
 
 @implementation TweetCell
 
 - (void)awakeFromNib {
     [super awakeFromNib];
-    // Initialization code
 }
 
 - (void)setSelected:(BOOL)selected animated:(BOOL)animated {
     [super setSelected:selected animated:animated];
+}
+
+- (void)refreshData:(Tweet *)tweet {
+    self.cellTweet = tweet;
+    [self refreshCurrentCell];
+}
+
+- (void)refreshCurrentCell {
+    NSString *URLString = self.cellTweet.user.profilePicture;
+    NSURL *url = [NSURL URLWithString:URLString];
+    NSData *urlData = [NSData dataWithContentsOfURL:url];
+    self.profilePicture.image = [UIImage imageWithData:urlData];
+    self.profilePicture.layer.cornerRadius = self.profilePicture.frame.size.width / 2;
+    
+    self.userDisplayName.text = self.cellTweet.user.screenName;
+    self.userAccountName.text = self.cellTweet.user.name;
+    
+    self.tweetText.text = self.cellTweet.text;
+    self.tweetText.linkTextAttributes = @{NSForegroundColorAttributeName:[UIColor systemBlueColor]};
+    self.tweetText.dataDetectorTypes = UIDataDetectorTypeLink;
+    
+    self.displayDate.text = [self.cellTweet.createdAtDate shortTimeAgoSinceNow];
+    
+    self.retweetCountLabel.text = [NSString stringWithFormat:@"%d", self.cellTweet.retweetCount];
+    self.likeCountLabel.text = [NSString stringWithFormat:@"%d", self.cellTweet.favoriteCount];
+    
+    [self updateRetweetFavorite];
+}
+
+- (void)updateRetweetFavorite {
+    if (self.cellTweet.retweeted) {
+        self.retweetButton.selected = true;
+    } else {
+        self.retweetButton.selected = false;
+    }
+    
+    if (self.cellTweet.favorited) {
+        self.favoriteButton.selected = true;
+    } else {
+        self.favoriteButton.selected = false;
+    }
+}
+
+- (IBAction)didTapFavorite:(id)sender {
```

Comment:
You can take this argument as far as to updating retweets and favorite status. How would this look if you used a delegate to the controller for these taps? How do you make the view as dumb as possible?